src/context.rs

```diff
@@ -225,10 +227,28 @@ impl Game {
                         }
                     }
                 }
+                ScheduledStop::Frame(count) => {
+                    if frame_ended {
+                        if *count == 1 {
+                            if !self.cpu.controller.is_instruction_finished {
+                                self.scheduled_stop = Some(ScheduledStop::Instruction(1));
+                            } else {
+                                self.emulation_stopped = true;
+                                self.scheduled_stop = None;
+                            }
```

Comment:
[code duplication] create a function `stop_here_or_at_the_end_of_opcode` (something like that)